pyvista/core/filters/data_set.py

```diff
@@ -5482,20 +5485,58 @@ def separate_cells(self):
         return self.shrink(1.0)
 
 
-def _set_threshold_limit(alg, value, invert):
-    """Set vtkThreshold limit.
+def _set_threshold_limit(alg, value, method, invert):
+    """Set vtkThreshold limits and function.
+
+    Addresses VTK API deprecations and previous PyVista inconsistencies with ParaView. Reference:
 
-    Addresses VTK API deprecation as pointed out in
-    https://github.com/pyvista/pyvista/issues/2850
+    * https://github.com/pyvista/pyvista/issues/2850
+    * https://github.com/pyvista/pyvista/issues/3610
+    * https://discourse.vtk.org/t/unnecessary-vtk-api-change/9929
 
     """
-    if invert:
-        if pyvista.vtk_version_info >= (9, 1):
-            alg.SetUpperThreshold(value)
-        else:  # pragma: no cover
-            alg.ThresholdByLower(value)
-    else:
-        if pyvista.vtk_version_info >= (9, 1):
-            alg.SetLowerThreshold(value)
-        else:  # pragma: no cover
-            alg.ThresholdByUpper(value)
+    # Check value
+    if isinstance(value, (np.ndarray, collections.abc.Sequence)):
+        if len(value) != 2:
+            raise ValueError(
+                f'Value range must be length one for a float value or two for min/max; not ({value}).'
+            )
+    elif isinstance(value, collections.abc.Iterable):
+        raise TypeError('Value must either be a single scalar or a sequence.')
+    if pyvista.vtk_version_info >= (9,):
+        alg.SetInvert(invert)
+    elif invert:  # pragma: no cover
+        raise ValueError('PyVista no longer supports inverted thresholds for VTK<9.')
+    # Check range
+    if isinstance(value, (np.ndarray, collections.abc.Sequence)) and value[0] > value[1]:
+        raise ValueError(
+            'Value sequence is invalid, please use (min, max). The provided first value is greater than the second.'
+        )
```

Comment:
Ehh... this might be cleaner to place under the previous `if isinstance(value, (np.ndarray, collections.abc.Sequence))` block